sydent/sydent.py

```diff
@@ -523,15 +305,8 @@ def run_gc():
 
 
 if __name__ == "__main__":
-    cfg = parse_config_file(get_config_file_path())
-    setup_logging(cfg)
-
     sydent_config = SydentConfig()
-    cfg_needs_saving = sydent_config.parse_from_config_parser(cfg)
-
-    syd = Sydent(cfg, sydent_config=sydent_config)
-
-    if cfg_needs_saving:
-        syd.save_config()
+    sydent_config.parse_config_file(get_config_file_path())
 
+    syd = Sydent(sydent_config)
```

Comment:
This commit (37b928b4) is a bit of a mammoth :/

I think you could have split it up like this:
 * move `save_config()` into `SydentConfig` (you can always inline it later if you want)
 * remove the `cfg` param from `Sydent.__init__`
 * move `CONFIG_DEFAULTS` to `config.__init__.py`
 * move `parse_config_file` and `parse_config_dict` into `SydentConfig`

Just a thought for the future!

Reply:
Ahh sorry I realised this was quite big, but thought it might be ok as I was just cut and pasting stuff (which I can now see is mainly easier to understand as the writer rather than as the reviewer!)